libraries/ArduinoOTA/examples/OTAConfigErase/WifiHealth.ino

```diff
@@ -0,0 +1,349 @@
+      r = F("ASSOC_TOOMANY");
+      break;
+
+    case WIFI_DISCONNECT_REASON_NOT_AUTHED:
+      r = F("NOT_AUTHED");
+      break;
+
+    case WIFI_DISCONNECT_REASON_NOT_ASSOCED:
+      r = F("NOT_ASSOCED");
+      break;
+
+    case WIFI_DISCONNECT_REASON_ASSOC_LEAVE:
+      r = F("ASSOC_LEAVE");
+      break;
+
+    case WIFI_DISCONNECT_REASON_ASSOC_NOT_AUTHED:
+      r = F("ASSOC_NOT_AUTHED");
+      break;
+
+    case WIFI_DISCONNECT_REASON_DISASSOC_PWRCAP_BAD:
+      r = F("DISASSOC_PWRCAP_BAD");
+      break;
+
+    case WIFI_DISCONNECT_REASON_DISASSOC_SUPCHAN_BAD:
+      r = F("DISASSOC_SUPCHAN_BAD");
+      break;
+
+    case WIFI_DISCONNECT_REASON_IE_INVALID:
+      r = F("IE_INVALID");
+      break;
+
+    case WIFI_DISCONNECT_REASON_MIC_FAILURE:
+      r = F("MIC_FAILURE");
+      break;
+
+    case WIFI_DISCONNECT_REASON_4WAY_HANDSHAKE_TIMEOUT:
+      r = F("4WAY_HANDSHAKE_TIMEOUT");
+      break;
+
+    case WIFI_DISCONNECT_REASON_GROUP_KEY_UPDATE_TIMEOUT:
+      r = F("GROUP_KEY_UPDATE_TIMEOUT");
+      break;
+
+    case WIFI_DISCONNECT_REASON_IE_IN_4WAY_DIFFERS:
+      r = F("IE_IN_4WAY_DIFFERS");
+      break;
+
+    case WIFI_DISCONNECT_REASON_GROUP_CIPHER_INVALID:
+      r = F("GROUP_CIPHER_INVALID");
+      break;
+
+    case WIFI_DISCONNECT_REASON_PAIRWISE_CIPHER_INVALID:
+      r = F("PAIRWISE_CIPHER_INVALID");
+      break;
+
+    case WIFI_DISCONNECT_REASON_AKMP_INVALID:
+      r = F("AKMP_INVALID");
+      break;
+
+    case WIFI_DISCONNECT_REASON_UNSUPP_RSN_IE_VERSION:
+      r = F("UNSUPP_RSN_IE_VERSION");
+      break;
+
+    case WIFI_DISCONNECT_REASON_INVALID_RSN_IE_CAP:
+      r = F("INVALID_RSN_IE_CAP");
+      break;
+
+    case WIFI_DISCONNECT_REASON_802_1X_AUTH_FAILED:
+      r = F("802_1X_AUTH_FAILED");
+      break;
+
+    case WIFI_DISCONNECT_REASON_CIPHER_SUITE_REJECTED:
+      r = F("CIPHER_SUITE_REJECTED");
+      break;
+
+    case WIFI_DISCONNECT_REASON_BEACON_TIMEOUT:
+      r = F("BEACON_TIMEOUT");
+      break;
+
+    case WIFI_DISCONNECT_REASON_NO_AP_FOUND:
+      r = F("NO_AP_FOUND");
+      break;
+
+    case WIFI_DISCONNECT_REASON_AUTH_FAIL:
+      r = F("AUTH_FAIL");
+      break;
+
+    case WIFI_DISCONNECT_REASON_ASSOC_FAIL:
+      r = F("ASSOC_FAIL");
+      break;
+
+    case WIFI_DISCONNECT_REASON_HANDSHAKE_TIMEOUT:
+      r = F("HANDSHAKE_TIMEOUT");
+      break;
+
+    default:
+      return String(F("Unknown: ")) + String(reason, HEX);
+  }
+  return String(r);
+}
+
+char getPhyModeChar(WiFiPhyMode_t i) {
+  switch (i) {
+    case WIFI_PHY_MODE_11B:
+      return 'b'; // = 1
+    case WIFI_PHY_MODE_11G:
+      return 'g'; // = 2,
+    case WIFI_PHY_MODE_11N:
+      return 'n'; // = 3,
+    default:
+      break;
+  }
+  return '?';
+}
+
+void updateWiFiStats(void) {
+  int32_t rssi = WiFi.RSSI();
+  if (rssi < 10) {
+    wifiHealth.rssi_max = max((int8_t)rssi, wifiHealth.rssi_max);
+    wifiHealth.rssi_min = min((int8_t)rssi, wifiHealth.rssi_min);
```

Comment:
Here an example of things that may to wrong if the wifiHealth members aren't properly initialized.